### PR TITLE
refactor(): builds, jobs running times

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: src/app/services/app.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Injectable, Provider } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class AppService {
```
Comment (jkuri) [inline src/app/services/app.service.ts]: I think `TimeService` would be more appropriate.

File: src/app/services/app.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  timer: Observable<any>;
+
+  constructor() {
+    this.timer = Observable.timer(1000, 1000).share();
```
Comment (jkuri) [inline src/app/services/app.service.ts]: you don't need to `.share()` inner observables (you are using that one in `getCurrentTime` which is shared already).

**Comment by irmana** [comment]
To get a 5 star review, I would love to see other time-related fixes being part of this PR. See Jan's last comment.

**Comment by Izak88** [comment]
I've added a correction of "time" commited about X ago, which wasn't working properly. About approximately remaining time and progress bars I didn't found a problem when I tested it all worked. Can you tell me what do you think is wrong with this two, where they don't work...

File: src/app/services/app.service.ts (hunk 1 above)
Comment (Izak88) [reply src/app/services/app.service.ts]: done

File: src/app/services/app.service.ts (hunk 2 above)
Comment (Izak88) [reply src/app/services/app.service.ts]: done

**Comment by jkuri** [comment]
check CI

**Comment by jkuri** [comment]
please dobule check progress bars on all screens. then i will approve.

**Comment by Izak88** [comment]
You're right, in some cases, progress bar didn't work. It's fixed now...

**Pull request merged**